Io500 sc 19

JulianKunkel opened the pull request:
Time to merge back

JulianKunkel committed:
Moved defaults into io500_fixed.sh
Set default stonewall to 1s.

JulianKunkel committed:
Checkout git tag for SC-19.

JulianKunkel committed:
Use option for MDtest to output time AND rate.

JulianKunkel committed:
Merge branch 'master' into io500-sc-19

JulianKunkel committed:
Use -X flag for verification in MDTest. Use sc19 tag for ior.

JulianKunkel committed:
Added changelog

JulianKunkel committed:
MDTest use accurate timer for reporting time

JulianKunkel committed:
Allow to optionally synchronize cache between benchmarks.

JulianKunkel committed:
Changelog upd.

JulianKunkel committed:
Extract the 300s limit, as this simplifies SCC adaption.

johnbent commented on CHANGELOG.md:
```diff
@@ -0,0 +1,9 @@
+  * IOR and MDTest now output the performance when reaching the stonewall
+  * MDTest uses -X flag and verifies that data read is matching the expected pattern
+  * MDTest now uses the new -P flag to print rate and time to ease debugging
+  * MDTest now uses rank shifting supporting block and cyclic layout (auto-detected)
```
IOR also now auto-detects block and cyclic.  It used to only do one of them.

JulianKunkel replied:
That is part of the changelog.md description and what we discussed at our meeting:
  * MDTest uses -X flag and verifies that data read is matching the expected pattern
  * MDTest now uses the new -P flag to print rate and time to ease debugging

johnbent commented on io500.sh:
```diff
@@ -137,31 +131,13 @@ function run_benchmarks {
-# Add key/value pairs defining your system
-# Feel free to add extra ones if you'd like
+# Information fields; these provide information about your system hardware
+# Use https://vi4io.org/io500-info-creator/ to generate information about your hardware
```
We said we wouldn't make this change until the info creator had pull-downs.

gmarkomanolis replied:
I am working on this, I have to fix some bugs.

johnbent commented on utilities/io500_fixed.sh:
```diff
@@ -5,29 +5,71 @@
 
+function set_defaults {
+  # Set default values, they may be overwritten by the user if necessary
+  io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
```
Can you please set this to 900000?  900000 is ten times the highest value that has ever been set.

johnbent commented on utilities/io500_fixed.sh:
```diff
@@ -5,29 +5,71 @@
+function set_defaults {
+  # Set default values, they may be overwritten by the user if necessary
+  io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
+  io500_mdtest_hard_files_per_proc=${io500_mdtest_hard_files_per_proc:-500000}
```
Can you please set this to 950000?  950000 is ten times the highest value that has ever been set.

johnbent commented on utilities/io500_fixed.sh:
```diff
@@ -5,29 +5,71 @@
+  # Set default values, they may be overwritten by the user if necessary
+  io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
+  io500_mdtest_hard_files_per_proc=${io500_mdtest_hard_files_per_proc:-500000}
+  io500_ior_hard_writes_per_proc=${io500_ior_hard_writes_per_proc:-100000}
```
95000 is ten times the highest value that has ever been seen.  Your number is already higher than that.  :)

johnbent commented on utilities/io500_fixed.sh:
```diff
@@ -5,29 +5,71 @@
+  # io500_ior_easy_size is the amount of data written per rank in MiB units,
+  # but it can be any number as long as it is somehow used to scale the IOR
+  # runtime as part of io500_ior_easy_params
+  io500_ior_easy_size=${io500_ior_easy_size:-200000}
```
9920000 please. [10 times max so far]

johnbent commented on utilities/io500_fixed.sh:
```diff
@@ -5,29 +5,71 @@
+}
+
+function drop_cache {
+  sudo bash -c "echo 3 > /proc/sys/vm/drop_caches"
```
What happens if this is run by regular user?  Won't this interrupt the script and ask for a password?

gmarkomanolis replied:
Basically the user needs to add an option to allow mpirun execution as root also (--allow-run-as-root). This is though bad practice but not sure what is the best thing here.

adilger replied:
sudo(8) shows:

     -n, --non-interactive
                   Avoid prompting the user for input of any kind.  If a
                   password is required for the command to run, sudo will
                   display an error message and exit.

so use `sudo -n` and append `|| true` at the end so that the failure to clear the cache is not considered an error.

johnbent commented on utilities/io500_fixed.sh:
```diff
@@ -164,29 +219,32 @@ function mdt_hard {
   [ "$io500_run_md_hard" != "True" ] && printf "\n[Skipping] $phase\n" && return 0
 
-  params_md_hard="-t -F -w $mdt_hard_fsize -e $mdt_hard_fsize -d $io500_workdir/mdt_hard -n $io500_mdtest_hard_files_per_proc -x $io500_workdir/mdt_hard-stonewall -a $io500_mdtest_hard_api $io500_mdtest_hard_api_specific_options -N 1"
+  params_md_hard="-t -F -P -w $mdt_hard_fsize -e $mdt_hard_fsize -d $io500_workdir/mdt_hard -n $io500_mdtest_hard_files_per_proc -x $io500_workdir/mdt_hard-stonewall -a $io500_mdtest_hard_api $io500_mdtest_hard_api_specific_options -N 1"
```
What does -P do to mdtest?  I don't see it in the Usage . . .

johnbent commented on utilities/io500_fixed.sh:
```diff
@@ -164,29 +219,32 @@ function mdt_hard {
     [ "$io500_run_md_hard_read" != "True" ] && printf "\n[Skipping] $phase\n" && return 0
     startphase
-    myrun "$io500_mdtest_cmd -E $params_md_hard" $result_file
+    myrun "$io500_mdtest_cmd -X -E $params_md_hard" $result_file
```
What does -X do?  I don't see it in the Usage.

johnbent reviewed (changes requested):
Hey Julian, looks very nice.  Can you please make a few small requested changes and answer the few questions I had?  Oh, also, I thought we agreed to not change to the info creator until it had the pull downs.  So let's please either get those pulldowns or remove the info creator for now.  Thanks.

JulianKunkel committed:
Updated default values.

JulianKunkel commented:
I think that one we cover by setting "mpiexec -np X ---allow-run-as-root"
could be specified maybe?

On Fri, Aug 23, 2019 at 6:11 PM George Markomanolis <
notifications@github.com> wrote:

> *@gmarkomanolis* commented on this pull request.
> ------------------------------
>
> In utilities/io500_fixed.sh
> <https://github.com/VI4IO/io-500-dev/pull/47#discussion_r317224383>:
>
> > +  # Set default values, they may be overwritten by the user if necessary
> +  io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
> +  io500_mdtest_hard_files_per_proc=${io500_mdtest_hard_files_per_proc:-500000}
> +  io500_ior_hard_writes_per_proc=${io500_ior_hard_writes_per_proc:-100000}
> +  # io500_ior_easy_size is the amount of data written per rank in MiB units,
> +  # but it can be any number as long as it is somehow used to scale the IOR
> +  # runtime as part of io500_ior_easy_params
> +  io500_ior_easy_size=${io500_ior_easy_size:-200000}
> +  # 2M writes, 2 GB per proc, file per proc
> +  io500_ior_easy_params=${io500_ior_easy_params:-"-t 2048k -b ${io500_ior_easy_size}m -F"}
> +  io500_mdreal_params=${io500_mdreal_params:-"-P=5000 -I=1000"}
> +  io500_clean_cache_cmd=${io500_clean_cache_cmd:-drop_cache}
> +}
> +
> +function drop_cache {
> +  sudo bash -c "echo 3 > /proc/sys/vm/drop_caches"
>
> Basically the user needs to add an option to allow mpirun execution as
> root also (--allow-run-as-root). This is though bad practice but not sure
> what is the best thing here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=ABGW5SRKODIIRSCULMIVIBDQGAK3XA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCRQXHA#discussion_r317224383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABGW5SXKBKFNTLP3KTVUJCDQGAK3XANCNFSM4IOZ3D4Q>
> .
>


-- 
Dr. Julian Kunkel
Lecturer, Department of Computer Science
+44 (0) 118 378 8218
http://www.cs.reading.ac.uk/
https://hps.vi4io.org/
PGP Fingerprint: 1468 1A86 A908 D77E B40F 45D6 2B15 73A5 9D39 A28E

johnbent commented:
I don't think we can ask the community to run this as root.  io500.sh has a
cleanup routine that does \rm -rf so if there is a bug, we just trashed
someone's production file system.  Also, many of the community do not have
sudo permissions.

On Fri, Aug 23, 2019 at 11:12 AM Julian Kunkel <notifications@github.com>
wrote:

> I think that one we cover by setting "mpiexec -np X ---allow-run-as-root"
> could be specified maybe?
>
> On Fri, Aug 23, 2019 at 6:11 PM George Markomanolis <
> notifications@github.com> wrote:
>
> > *@gmarkomanolis* commented on this pull request.
> > ------------------------------
> >
> > In utilities/io500_fixed.sh
> > <https://github.com/VI4IO/io-500-dev/pull/47#discussion_r317224383>:
> >
> > > + # Set default values, they may be overwritten by the user if
> necessary
> > +
> io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
> > +
> io500_mdtest_hard_files_per_proc=${io500_mdtest_hard_files_per_proc:-500000}
> > +
> io500_ior_hard_writes_per_proc=${io500_ior_hard_writes_per_proc:-100000}
> > + # io500_ior_easy_size is the amount of data written per rank in MiB
> units,
> > + # but it can be any number as long as it is somehow used to scale the
> IOR
> > + # runtime as part of io500_ior_easy_params
> > + io500_ior_easy_size=${io500_ior_easy_size:-200000}
> > + # 2M writes, 2 GB per proc, file per proc
> > + io500_ior_easy_params=${io500_ior_easy_params:-"-t 2048k -b
> ${io500_ior_easy_size}m -F"}
> > + io500_mdreal_params=${io500_mdreal_params:-"-P=5000 -I=1000"}
> > + io500_clean_cache_cmd=${io500_clean_cache_cmd:-drop_cache}
> > +}
> > +
> > +function drop_cache {
> > + sudo bash -c "echo 3 > /proc/sys/vm/drop_caches"
> >
> > Basically the user needs to add an option to allow mpirun execution as
> > root also (--allow-run-as-root). This is though bad practice but not sure
> > what is the best thing here.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=ABGW5SRKODIIRSCULMIVIBDQGAK3XA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCRQXHA#discussion_r317224383
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/ABGW5SXKBKFNTLP3KTVUJCDQGAK3XANCNFSM4IOZ3D4Q
> >
> > .
> >
>
>
> --
> Dr. Julian Kunkel
> Lecturer, Department of Computer Science
> +44 (0) 118 378 8218
> http://www.cs.reading.ac.uk/
> https://hps.vi4io.org/
> PGP Fingerprint: 1468 1A86 A908 D77E B40F 45D6 2B15 73A5 9D39 A28E
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=AAPT2PQD3ZCK3IKDN5XT4CLQGALBVA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5AZUDQ#issuecomment-524392974>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPT2PU3NK2JDOKN4WMUVU3QGALBVANCNFSM4IOZ3D4Q>
> .
>

JulianKunkel commented:
Both of these features are to support testcases and stuff like the student
cluster competition. Yes they shall never be run on a production system.
And they couldn't because you wouldn't have sudo rights or actually run as
root....

On Fri, Aug 23, 2019 at 6:27 PM John Bent <notifications@github.com> wrote:

> I don't think we can ask the community to run this as root. io500.sh has a
> cleanup routine that does \rm -rf so if there is a bug, we just trashed
> someone's production file system. Also, many of the community do not have
> sudo permissions.
>
> On Fri, Aug 23, 2019 at 11:12 AM Julian Kunkel <notifications@github.com>
> wrote:
>
> > I think that one we cover by setting "mpiexec -np X ---allow-run-as-root"
> > could be specified maybe?
> >
> > On Fri, Aug 23, 2019 at 6:11 PM George Markomanolis <
> > notifications@github.com> wrote:
> >
> > > *@gmarkomanolis* commented on this pull request.
> > > ------------------------------
> > >
> > > In utilities/io500_fixed.sh
> > > <https://github.com/VI4IO/io-500-dev/pull/47#discussion_r317224383>:
> > >
> > > > + # Set default values, they may be overwritten by the user if
> > necessary
> > > +
> >
> io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
> > > +
> >
> io500_mdtest_hard_files_per_proc=${io500_mdtest_hard_files_per_proc:-500000}
> > > +
> > io500_ior_hard_writes_per_proc=${io500_ior_hard_writes_per_proc:-100000}
> > > + # io500_ior_easy_size is the amount of data written per rank in MiB
> > units,
> > > + # but it can be any number as long as it is somehow used to scale the
> > IOR
> > > + # runtime as part of io500_ior_easy_params
> > > + io500_ior_easy_size=${io500_ior_easy_size:-200000}
> > > + # 2M writes, 2 GB per proc, file per proc
> > > + io500_ior_easy_params=${io500_ior_easy_params:-"-t 2048k -b
> > ${io500_ior_easy_size}m -F"}
> > > + io500_mdreal_params=${io500_mdreal_params:-"-P=5000 -I=1000"}
> > > + io500_clean_cache_cmd=${io500_clean_cache_cmd:-drop_cache}
> > > +}
> > > +
> > > +function drop_cache {
> > > + sudo bash -c "echo 3 > /proc/sys/vm/drop_caches"
> > >
> > > Basically the user needs to add an option to allow mpirun execution as
> > > root also (--allow-run-as-root). This is though bad practice but not
> sure
> > > what is the best thing here.
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=ABGW5SRKODIIRSCULMIVIBDQGAK3XA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCRQXHA#discussion_r317224383
> > >,
> > > or mute the thread
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ABGW5SXKBKFNTLP3KTVUJCDQGAK3XANCNFSM4IOZ3D4Q
> > >
> > > .
> > >
> >
> >
> > --
> > Dr. Julian Kunkel
> > Lecturer, Department of Computer Science
> > +44 (0) 118 378 8218
> > http://www.cs.reading.ac.uk/
> > https://hps.vi4io.org/
> > PGP Fingerprint: 1468 1A86 A908 D77E B40F 45D6 2B15 73A5 9D39 A28E
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=AAPT2PQD3ZCK3IKDN5XT4CLQGALBVA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5AZUDQ#issuecomment-524392974
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AAPT2PU3NK2JDOKN4WMUVU3QGALBVANCNFSM4IOZ3D4Q
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=ABGW5SVGFHDDEIYNED3JRYLQGAMXVA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5A2WUY#issuecomment-524397395>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABGW5SW3INTY3WCPN6VH2PDQGAMXVANCNFSM4IOZ3D4Q>
> .
>


-- 
Dr. Julian Kunkel
Lecturer, Department of Computer Science
+44 (0) 118 378 8218
http://www.cs.reading.ac.uk/
https://hps.vi4io.org/
PGP Fingerprint: 1468 1A86 A908 D77E B40F 45D6 2B15 73A5 9D39 A28E

johnbent commented:
I suppose if the default is to keep it turned off, then maybe it's OK.
Curious to hear whether George and Jay are comfortable with this.

On Fri, Aug 23, 2019 at 11:28 AM Julian Kunkel <notifications@github.com>
wrote:

> Both of these features are to support testcases and stuff like the student
> cluster competition. Yes they shall never be run on a production system.
> And they couldn't because you wouldn't have sudo rights or actually run as
> root....
>
> On Fri, Aug 23, 2019 at 6:27 PM John Bent <notifications@github.com>
> wrote:
>
> > I don't think we can ask the community to run this as root. io500.sh has
> a
> > cleanup routine that does \rm -rf so if there is a bug, we just trashed
> > someone's production file system. Also, many of the community do not have
> > sudo permissions.
> >
> > On Fri, Aug 23, 2019 at 11:12 AM Julian Kunkel <notifications@github.com
> >
> > wrote:
> >
> > > I think that one we cover by setting "mpiexec -np X
> ---allow-run-as-root"
> > > could be specified maybe?
> > >
> > > On Fri, Aug 23, 2019 at 6:11 PM George Markomanolis <
> > > notifications@github.com> wrote:
> > >
> > > > *@gmarkomanolis* commented on this pull request.
> > > > ------------------------------
> > > >
> > > > In utilities/io500_fixed.sh
> > > > <https://github.com/VI4IO/io-500-dev/pull/47#discussion_r317224383>:
> > > >
> > > > > + # Set default values, they may be overwritten by the user if
> > > necessary
> > > > +
> > >
> >
> io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
> > > > +
> > >
> >
> io500_mdtest_hard_files_per_proc=${io500_mdtest_hard_files_per_proc:-500000}
> > > > +
> > >
> io500_ior_hard_writes_per_proc=${io500_ior_hard_writes_per_proc:-100000}
> > > > + # io500_ior_easy_size is the amount of data written per rank in MiB
> > > units,
> > > > + # but it can be any number as long as it is somehow used to scale
> the
> > > IOR
> > > > + # runtime as part of io500_ior_easy_params
> > > > + io500_ior_easy_size=${io500_ior_easy_size:-200000}
> > > > + # 2M writes, 2 GB per proc, file per proc
> > > > + io500_ior_easy_params=${io500_ior_easy_params:-"-t 2048k -b
> > > ${io500_ior_easy_size}m -F"}
> > > > + io500_mdreal_params=${io500_mdreal_params:-"-P=5000 -I=1000"}
> > > > + io500_clean_cache_cmd=${io500_clean_cache_cmd:-drop_cache}
> > > > +}
> > > > +
> > > > +function drop_cache {
> > > > + sudo bash -c "echo 3 > /proc/sys/vm/drop_caches"
> > > >
> > > > Basically the user needs to add an option to allow mpirun execution
> as
> > > > root also (--allow-run-as-root). This is though bad practice but not
> > sure
> > > > what is the best thing here.
> > > >
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> > >
> >
> https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=ABGW5SRKODIIRSCULMIVIBDQGAK3XA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCRQXHA#discussion_r317224383
> > > >,
> > > > or mute the thread
> > > > <
> > >
> >
> https://github.com/notifications/unsubscribe-auth/ABGW5SXKBKFNTLP3KTVUJCDQGAK3XANCNFSM4IOZ3D4Q
> > > >
> > > > .
> > > >
> > >
> > >
> > > --
> > > Dr. Julian Kunkel
> > > Lecturer, Department of Computer Science
> > > +44 (0) 118 378 8218
> > > http://www.cs.reading.ac.uk/
> > > https://hps.vi4io.org/
> > > PGP Fingerprint: 1468 1A86 A908 D77E B40F 45D6 2B15 73A5 9D39 A28E
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=AAPT2PQD3ZCK3IKDN5XT4CLQGALBVA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5AZUDQ#issuecomment-524392974
> > >,
> > > or mute the thread
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AAPT2PU3NK2JDOKN4WMUVU3QGALBVANCNFSM4IOZ3D4Q
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=ABGW5SVGFHDDEIYNED3JRYLQGAMXVA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5A2WUY#issuecomment-524397395
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/ABGW5SW3INTY3WCPN6VH2PDQGAMXVANCNFSM4IOZ3D4Q
> >
> > .
> >
>
>
> --
> Dr. Julian Kunkel
> Lecturer, Department of Computer Science
> +44 (0) 118 378 8218
> http://www.cs.reading.ac.uk/
> https://hps.vi4io.org/
> PGP Fingerprint: 1468 1A86 A908 D77E B40F 45D6 2B15 73A5 9D39 A28E
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=AAPT2PQUGGMF35DEMQMNP53QGAM47A5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5A22BA#issuecomment-524397828>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPT2PU5AGKIKWM6VBHZSHLQGAM47ANCNFSM4IOZ3D4Q>
> .
>

gmarkomanolis commented:
> I suppose if the default is to keep it turned off, then maybe it's OK. Curious to hear whether George and Jay are comfortable with this. On Fri, Aug 23, 2019 at 11:28 AM Julian Kunkel <notifications@github.com> wrote:
> […](#)

This is only for SCC and similar, never for a production. We use $PWD for path so I hope, it is difficult to damage something. If this command would help to have more realistic results, it's good. I will try to test if possible on personal computer. 
Yes, the default is off.

gmarkomanolis committed:
Update io500.sh

Adding scc flag, if it is 0, IO500 runs as expected, if it is 1, then the threshold is 30 seconds.

gmarkomanolis committed:
Update io500_fixed.sh

if scc==0 then threshold 300 seconds, if scc==1, 30 seconds of threshold.

gmarkomanolis commented:
> Is this variable referenced in io500_fixed.sh?

It is used by io500_fixed to decide to use 300s or 30s. I use numbers in case in the future we have another case e.g. 2 where if 2 then do that.

JulianKunkel committed:
Adilger's comment use -n.

gmarkomanolis committed:
Update io500.sh

change variable scc to io500_rules

gmarkomanolis committed:
Update io500_fixed.sh

adapt to io500_rules variable

JulianKunkel committed:
Bail out if no appropriate io500_rules set.

johnbent commented:
APPROVE.

gmarkomanolis commented:
APPROVE

johnbent commented:
APPROVE.

On Mon, Aug 26, 2019 at 9:06 AM George Markomanolis <
notifications@github.com> wrote:

> APPROVE
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/VI4IO/io-500-dev/pull/47?email_source=notifications&email_token=AAPT2PU7QRVWLJDWQYRMMPLQGPWQFA5CNFSM4IOZ3D42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5EU63I#issuecomment-524898157>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPT2PWDLRGPZLAJFNMJL2TQGPWQFANCNFSM4IOZ3D4Q>
> .
>
